packages/flutter/lib/src/material/page.dart

```diff
@@ -85,59 +96,37 @@ class MaterialPageRoute<T> extends PageRoute<T> {
 
   @override
   bool canTransitionFrom(TransitionRoute<dynamic> nextRoute) {
-    return nextRoute is MaterialPageRoute<dynamic>;
+    return nextRoute is MaterialPageRoute<dynamic> || nextRoute is CupertinoPageRoute<dynamic>;
```

Comment:
alternatively, you could say `nextRoute is MaterialPageRoute<dynamic> || (_cupertinoPageRoute != null && _cupertinoPageRoute.canTransitionFrom(nextRoute))` or some such?

Same below.